pkg/fanal/analyzer/pkg/apk/apk_test.go

```diff
@@ -6,302 +6,426 @@ import (
+	"usr/lib/engines/libcapi.so",
+	"usr/lib/engines/libgost.so",
+	"usr/lib/engines/libcswift.so",
+	"usr/lib/engines/libchil.so",
+	"usr/lib/engines/libgmp.so",
+	"usr/lib/engines/libnuron.so",
+	"usr/lib/engines/lib4758cca.so",
+	"usr/lib/engines/libsureware.so",
+	"usr/lib/engines/libpadlock.so",
+	"usr/lib/engines/libaep.so",
 
-				// libssl1.0-1.0.2h-r1
-				"lib/libssl.so.1.0.0",
-				"usr/lib/libssl.so.1.0.0",
+	// libssl1.0-1.0.2h-r1
+	"lib/libssl.so.1.0.0",
+	"usr/lib/libssl.so.1.0.0",
 
-				// apk-tools-2.6.7-r0
-				"sbin/apk",
+	// apk-tools-2.6.7-r0
+	"sbin/apk",
 
-				// scanelf-1.1.6-r0
-				"usr/bin/scanelf",
+	// scanelf-1.1.6-r0
+	"usr/bin/scanelf",
 
-				// musl-utils-1.1.14-r10
-				"sbin/ldconfig",
-				"usr/bin/iconv",
-				"usr/bin/ldd",
-				"usr/bin/getconf",
-				"usr/bin/getent",
+	// musl-utils-1.1.14-r10
+	"sbin/ldconfig",
+	"usr/bin/iconv",
+	"usr/bin/ldd",
+	"usr/bin/getconf",
+	"usr/bin/getent",
 
-				// libc-utils-0.7-r0
+	// libc-utils-0.7-r0
 
-				// pkgconf-1.6.0-r0
-				"usr/bin/pkgconf",
-				"usr/bin/pkg-config",
-				"usr/lib/libpkgconf.so.3.0.0",
-				"usr/lib/libpkgconf.so.3",
-				"usr/share/aclocal/pkg.m4",
+	// pkgconf-1.6.0-r0
+	"usr/bin/pkgconf",
+	"usr/bin/pkg-config",
+	"usr/lib/libpkgconf.so.3.0.0",
+	"usr/lib/libpkgconf.so.3",
+	"usr/share/aclocal/pkg.m4",
 
-				// sqlite-libs-3.26.0-r3
-				"usr/lib/libsqlite3.so.0",
-				"usr/lib/libsqlite3.so.0.8.6",
+	// sqlite-libs-3.26.0-r3
+	"usr/lib/libsqlite3.so.0",
+	"usr/lib/libsqlite3.so.0.8.6",
 
-				// test-2.9.11_pre20061021-r2
-				"usr/lib/libsqlite3.so",
-				"usr/lib/pkgconfig/sqlite3.pc",
-				"usr/include/sqlite3ext.h",
-				"usr/include/sqlite3.h",
-			},
+	// test-2.9.11_pre20061021-r2
+	"usr/lib/libsqlite3.so",
+	"usr/lib/pkgconfig/sqlite3.pc",
+	"usr/include/sqlite3ext.h",
+	"usr/include/sqlite3.h",
+}
+
+func TestParseApkInfo(t *testing.T) {
+	var tests = map[string]struct {
+		path                       string
+		wantPkgs                   []types.Package
+		wantFiles                  []string
+		RetainSystemInstalledFiles bool
+	}{
+		"Valid": {
+			path:                       "./testdata/apk",
+			RetainSystemInstalledFiles: true,
+			wantPkgs:                   pkgs,
+			wantFiles:                  files,
+		},
+		"do not retain system installed files": {
+			path:                       "./testdata/apk",
+			RetainSystemInstalledFiles: false,
+			wantPkgs:                   pkgs,
+			wantFiles:                  files,
 		},
 	}
-	a := alpinePkgAnalyzer{}
-	for testname, v := range tests {
-		read, err := os.Open(v.path)
-		if err != nil {
-			t.Errorf("%s : can't open file %s", testname, v.path)
-		}
-		scanner := bufio.NewScanner(read)
-		gotPkgs, gotFiles := a.parseApkInfo(scanner)
-		assert.Equal(t, v.wantPkgs, gotPkgs)
-		assert.Equal(t, v.wantFiles, gotFiles)
+	for testname, tt := range tests {
+		t.Run(testname, func(t *testing.T) {
+			a := alpinePkgAnalyzer{}
+			read, err := os.Open(tt.path)
```

Comment:
This mistake is not from your change, but we should close this file.

```suggestion
			f, err := os.Open(tt.path)
			defer f.Close()
```